### PR TITLE
chore: ensure module deps in ct-core

### LINKED ISSUE #23823: [BUG] @playwright/experimental-ct-react doesn't work out of the box with Yarn 3.6 PnP
### System info
- Playwright Version: "@playwright/experimental-ct-react": "^1.35.1"
- Operating System: Windows 11
- Browser: All
- Other info: Yarn v3.6 with PnP, Node v20.3.0

I followed https://playwright.dev/docs/test-components#how-to-get-started in order to add Playwright Component Testing support to my project. While all of the initialization steps seemed to succeed, running the test command fails.

### Source code

- [X] I provided exact source code that allows reproducing the issue locally.
[CommuteApp.zip](https://github.com/microsoft/playwright/files/11808510/CommuteApp.zip)

**Steps**
- Unzip provided source code
- Run `yarn`
- Run `yarn test:components`

**Expected**

Playwright test suite runs and denotes that the App.spec test failed as it's looking for different text than what's being rendered within App

**Actual**

Playwright test suite fails to start with the following error:
image

### PR DESCRIPTION
**Pull request by pavelfeldman**
Fixes https://github.com/microsoft/playwright/issues/23823

### CODE REVIEW

File: tests/components/ct-react-vite/package.json [hunk 1]
```diff
@@ -19,9 +19,5 @@
     "@vitejs/plugin-react": "^3.1.0",
     "typescript": "^4.5.4",
     "vite": "^4.2.1"
-  },
-  "@standaloneDevDependencies": {
```
Comment (aslushnikov) [inline tests/components/ct-react-vite/package.json:23]: what was this?

File: utils/check_deps.js [hunk 2]
```diff
@@ -209,22 +219,16 @@ async function innerCheckDeps(root) {
 
   function allowExternalImport(importName, packageJSON) {
     // Only external imports are relevant. Files in src/web are bundled via webpack.
-    if (importName.startsWith('.') || importName.startsWith('@'))
+    if (importName.startsWith('.') || (importName.startsWith('@') && !importName.startsWith('@playwright/')))
       return true;
     if (peerDependencies.includes(importName))
       return true;
-    try {
```
Comment (aslushnikov) [inline utils/check_deps.js:216]: why did this go away?

**Review by aslushnikov — APPROVED** [review]

File: tests/components/ct-react-vite/package.json (hunk 1 above)
Comment (pavelfeldman) [reply tests/components/ct-react-vite/package.json:23]: This was a hack to give user a hint how to run it in a standalone (non monorepo) mode.

File: utils/check_deps.js (hunk 2 above)
Comment (pavelfeldman) [reply utils/check_deps.js:216]: not needed anymore - we always resolve successfully in this code branch now

**Pull request merged**